ethereum/src/ERC20Treasury.sol

```diff
@@ -32,41 +32,39 @@ contract ERC20Treasury is IERC20Treasury, IMessageQueueReceiver {
         emit Deposit(tx.origin, to, token, amount);
```

Comment:
Why is tx.origin used here and also in the contract ERC20Gateway?
Using tx.origin can introduce security risks, particularly with intermediary contracts that could exploit this function.

Reply:
#193